guide/popular-topics/threads.md

```diff
@@ -0,0 +1,157 @@
+
+```js {3}
+client.on('messageCreate', async message => {
+	const thread = message.channel.threads.cache.find(x => x.name === 'food-talk');
+	if (thread.joinable) await thread.join();
+});
+```
+
+And to leave one you call `ThreadChannel#leave()`;
+
+```js {3}
+client.on('messageCreate', async message => {
+	const thread = message.channel.threads.cache.find(x => x.name === 'food-talk');
+	await thread.leave();
+});
+```
+
+## Archiving, unarchiving and locking threads
+
+Every thread can be either active or archived. Changing a thread from archived to active is referred to as unarchiving the thread. Threads that have `locked` set to true can only be unarchived by a member with the `MANAGE_THREADS` permission.
+
+Threads automatically archive after inactivity. 'Activity' is defined as sending a message, unarchiving a thread, or changing the auto-archive time.
+
+To archive a thread you use the `ThreadChannel#setArchived()` method and pass `true` as parameter:
+
+```js {3}
+client.on('messageCreate', async message => {
+	const thread = message.channel.threads.cache.find(x => x.name === 'food-talk');
+	await thread.setArchived(true);
+});
+```
+
+And to unarchive it one you pass `false ` to one and the same method:
+
+```js {3}
+client.on('messageCreate', async message => {
+	const thread = message.channel.threads.cache.find(x => x.name === 'food-talk');
+	await thread.setArchived(false);
+});
+```
+
+This same principle applies to locking and unlocking a thread via the `ThreadChannel#setLocked()` method:
+
+Locking a thread:
+
+```js {3}
+client.on('messageCreate', async message => {
+	const thread = message.channel.threads.cache.find(x => x.name === 'food-talk');
+	await thread.setLocked(true);
+});
+```
+
+::: warning
+Archived threads can't be locked!
+:::
+
+Unlocking a thread:
+
+```js {3}
+client.on('messageCreate', async message => {
+	const thread = message.channel.threads.cache.find(x => x.name === 'food-talk');
+	await thread.setLocked(false);
+});
+```
+
+## Public and private threads
+
+Public threads are viewable by everyone who can view the parent channel of the thread. Public threads must be created from an existing message, but can be 'orphaned' if that message is deleted. The created thread and the message it was started from will share the same id. The type of thread created matches the type of the parent channel.
+
+Private threads behave similar to Group DMs, but in a Guild. Private threads can only be created on text channels.
+
+To create a private thread you use `ThreadManager#create()` as above, but pass `private_thread` as `type`, like so:
+
+```js {6}
+client.on('messageCreate', async message => {
+	const thread = await message.channel.threads
+		.create({
+			name: 'mod-talk',
+			autoArchiveDuration: 60,
+			type: 'private_thread',
+			reason: 'Needed a separate thread for moderation',
+		});
+	console.log(`Created thread: ${thread.name}`);
+});
```

Comment:
```suggestion
To create a private thread, use `ThreadManager#create()` and pass in `private_thread` as the `type`:

```js {4}
const thread = await message.channel.threads.create({
	name: 'mod-talk',
	autoArchiveDuration: 60,
	type: 'private_thread',
	reason: 'Needed a separate thread for moderation',
});

console.log(`Created thread: ${thread.name}`);
```